### PR TITLE
chore(sdk): Support loading v1 yaml with {executorInput: } placeholder

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
We used to support `{executorInput: null}` placeholder (as a dictionary format) in v1 yaml, which is equivalent to the string value `"{{$}}"`, e.g.: https://github.com/kubeflow/pipelines/blob/e7dbbcb3b43006fc60c335f814b87b776f4cee96/components/google-cloud/google_cloud_pipeline_components/experimental/tensorflow_probability/anomaly_detection/component.yaml#L980

Just add the support so that it's backward compatible.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by connor-mccarthy** [comment]
Thank you, @chensun. I think this is going to have conflicting changes with #7795. Is is possible to merge #7795 first?

**Comment by chensun** [comment]
> Thank you, @chensun. I think this is going to have conflicting changes with #7795. Is is possible to merge #7795 first?

Sure, that's a rather large one, let's merge yours first.

**Comment by chensun** [comment]
> > Thank you, @chensun. I think this is going to have conflicting changes with #7795. Is is possible to merge #7795 first?
> 
> Sure, that's a rather large one, let's merge yours first.

Rebased on top of HEAD, please take a look @connor-mccarthy

File: sdk/python/kfp/components/placeholders.py [hunk 1]
```diff
@@ -146,6 +146,16 @@ def to_placeholder_string(self) -> str:
         return self._TO_PLACEHOLDER.format(**self.to_dict())
 
 
+class ExecutorInputPlaceholder(base_model.BaseModel,
+                               RegexPlaceholderSerializationMixin):
+    """Class that represents executor input placeholder."""
+    _TO_PLACEHOLDER = "{{$}}"
```
Comment (connor-mccarthy) [inline sdk/python/kfp/components/placeholders.py:152]: nit: prefer single quotes when no double quote character is contained inside the string

(throughout)

**Review by connor-mccarthy — APPROVED** [review]
Thanks, Chen!

/lgtm

File: sdk/python/kfp/components/placeholders.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/components/placeholders.py]: Done

**Comment by chensun** [comment]
The sample test failure is unrelated to this PR, and likely no sample in that test suite covers the change of this PR. 
Will follow up with the test issue separately in https://github.com/kubeflow/pipelines/issues/7893

**Pull request merged**